src/zkevm_specs/copy_circuit.py

```diff
@@ -64,28 +65,28 @@ def verify_step(cs: ConstraintSystem, rows: Sequence[CopyCircuitRow], r: FQ):
         cs.constrain_zero(rows[1].is_last * (1 - rows[0].bytes_left))
         # bytes_left == bytes_left_next + 1 for non-last step
         cs.constrain_zero((1 - rows[1].is_last) * (rows[0].bytes_left - rows[2].bytes_left - 1))
-
-        # write value == read value if not rlc accumulator
-        with cs.condition(1 - rows[1].is_rlc_acc):
-            cs.constrain_equal(rows[0].value, rows[1].value)
-        # read value == write value for the first step (always)
-        with cs.condition(rows[0].is_first):
-            cs.constrain_equal(rows[0].value, rows[1].value)
-
         # value == 0 when is_pad == 1 for read
         cs.constrain_zero(rows[0].is_pad * rows[0].value)
         # is_pad == 1 - (src_addr < src_addr_end) for read row
-        cs.constrain_equal(
-            1 - lt(rows[0].addr, rows[0].src_addr_end, N_BYTES_MEMORY_ADDRESS), rows[0].is_pad
-        )
+        # We skip tx_log because:
+        # 1. It can only ever be a write row, so q_step == 0 and the constraint will be satisfied.
+        # 2. Since `lt(..)` will still be computed, it's excepted to throw an exception since
+        #    dst addr is a very large number: addr += (int(TxLogFieldTag.Data) << 32) + (log_id << 48)
+        if rows[0].is_tx_log == FQ.zero():
```

Comment:
Why don't we use `with cs.condition(rows[0].is_tx_log):` here

Reply:
Yes, I've documented the reason on top.

If we use:
```
with cs.condition(cond.expr()) as cs:
    cs.constrain_equal(foo(), bar.expr())
```

then the `foo()` is still computed (even if `cond` is `0`), just that the constraint will be satisfied.

In the case of `TxLog`, the `addr` is a very large number:
```
addr += (int(TxLogFieldTag.Data) << 32) + (log_id << 48)
```

And the [`lt` function](https://github.com/scroll-tech/zkevm-specs/blob/fix/cs/src/zkevm_specs/copy_circuit.py#L16-L20) would panic at the assertion failure, since it's only meant to allow up to `N_BYTES_MEMORY_ADDRESS = 5` bytes (it's meant to check `addr < src_addr_end` kind of a constraint).

Hence I used an `if` instead of adding a `cs.condition`

Reply:
Oh, I see. What do you think we change the code comment from `We skip tx_log because:` to ` We use an if instead of adding a cs.condition`?